packages/mobx-state-tree/src/utils.ts

```diff
@@ -483,3 +483,30 @@ export function setImmediateWithFallback(fn: (...args: any[]) => void) {
         setTimeout(fn, 1)
     }
 }
+
+/**
+ * Convert a promise-returning function to a generator-returning one.
+ * This is intended to allow for usage of `yield*` in async actions to
+ * retain the promise return type.
+ *
+ * Example:
+ * ```ts
+ * function getDataAsync(input: string): Promise<number> { ... }
+ * const getDataGen = toGenerator(getDataAsync);
+ *
+ * const someModel.actions(self => ({
+ *   someAction: flow(function*() {
+ *     // value is typed as number
+ *     const value = yield* getDataGen("input value");
+ *     ...
+ *   })
+ * }))
+ * ```
+ */
+export function toGenerator<R, Args extends any[]>(
```

Comment:
Let's move it next to?
https://github.com/mobxjs/mobx-state-tree/blob/master/packages/mobx-state-tree/src/core/flow.ts#L19-L28
feels contextually more sound

Reply:
Makes sense - done.